src/FileStorage.php

```diff
@@ -49,28 +47,26 @@
     private $options;
 
     /**
-     * @var MessageCatalogue[] Fetched catalogies
+     * @var MessageCatalogue[] Fetched catalogues
      */
     private $catalogues;
 
     /**
-     * @param TranslationWriter                                            $writer
-     * @param SymfonyTranslationLoader|TranslationLoader|TranslationReader $loader
-     * @param array                                                        $dir
-     * @param array                                                        $options
+     * @param TranslationWriterInterface $writer
+     * @param TranslationReaderInterface $reader
+     * @param array                      $dir
+     * @param array                      $options
      */
-    public function __construct(TranslationWriter $writer, $loader, array $dir, array $options = [])
+    public function __construct($writer, $reader, array $dir, array $options = [])
     {
-        // Create a legacy writer which is a wrapper for TranslationWriter
+        // Create a wrapper for legacy writer
         if (!$writer instanceof TranslationWriterInterface) {
             $writer = new LegacyTranslationWriter($writer);
         }
-        // Create a legacy loader which is a wrapper for TranslationReader
-        if ($loader instanceof TranslationReader) {
-            $loader = new LegacyTranslationLoader($loader);
-        }
-        if (!$loader instanceof SymfonyTranslationLoader && !$loader instanceof TranslationLoader) {
-            throw new \LogicException('Second parameter of FileStorage must be a Symfony translation loader or implement Translation\SymfonyStorage\TranslationLoader');
+
+        // Create a wrapper for legacy reader
+        if (!$reader instanceof TranslationReaderInterface) {
+            $reader = new LegacyTranslationReader($reader);
```

Comment:
Nice idea to invert my logic with `LegacyTranslationReader` 👍 

I somehow didn't notice this way